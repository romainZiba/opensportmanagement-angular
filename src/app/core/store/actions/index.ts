export * from './layout.actions';
export * from './teams.actions';
