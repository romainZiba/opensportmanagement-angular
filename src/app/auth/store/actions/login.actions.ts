import {Action} from '@ngrx/store';
import {Credentials, User} from '../../models/user';

export enum LoginActionsType {
  LOGIN = '[Auth] Login',
  LOGIN_SUCCESS = '[Auth] Login Success',
  LOGIN_FAILURE = '[Auth] Login Failure',
  LOGIN_REDIRECT = '[Auth] Login Redirect',
  LOGOUT = '[Auth] Logout',
  CHECK_TOKEN_VALID = '[Auth] Check token valid'
}

export class Login implements Action {
  readonly type = LoginActionsType.LOGIN;
  constructor(public payload: Credentials) {}
}

export class LoginSuccess implements Action {
  readonly type = LoginActionsType.LOGIN_SUCCESS;
  constructor(public payload: User) {}
}

export class LoginFailure implements Action {
  readonly type = LoginActionsType.LOGIN_FAILURE;
  constructor(public payload: any) {}
}

export class LoginRedirect implements Action {
  readonly type = LoginActionsType.LOGIN_REDIRECT;
}

export class Logout implements Action {
  readonly type = LoginActionsType.LOGOUT;
}

export class CheckTokenValid implements Action {
  readonly type = LoginActionsType.CHECK_TOKEN_VALID
}

export type LoginActions =
  | Login
  | LoginSuccess
  | LoginFailure
  | LoginRedirect
  | Logout
  | CheckTokenValid;
