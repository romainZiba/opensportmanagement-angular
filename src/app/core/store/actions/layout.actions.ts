import { Action } from '@ngrx/store';

export enum SidenavActionsType {
  OPEN_SIDENAV = '[Core] Open Sidenav',
  CLOSE_SIDENAV = '[Core] Close Sidenav',
}

export class OpenSidenav implements Action {
  readonly type = SidenavActionsType.OPEN_SIDENAV;
}

export class CloseSidenav implements Action {
  readonly type = SidenavActionsType.CLOSE_SIDENAV;
}

export type SidenavActions = OpenSidenav | CloseSidenav;
