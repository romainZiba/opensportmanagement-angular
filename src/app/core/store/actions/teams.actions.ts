import { Action } from "@ngrx/store";
import { List } from "immutable";
import { Team } from "../../../model/team";

export enum TeamsActionsType {
  LOAD_TEAMS = "[Core] Load teams",
  LOAD_TEAMS_SUCCESS = "[Core] Load teams success",
  LOAD_TEAMS_FAILED = "[Core] Load teams failed",
  SELECT_TEAM = "[Core] Select team"
}

export class LoadTeams implements Action {
  readonly type = TeamsActionsType.LOAD_TEAMS;
}

export class LoadTeamsSuccess implements Action {
  readonly type = TeamsActionsType.LOAD_TEAMS_SUCCESS;
  constructor(public payload: List<Team>) {}
}

export class LoadTeamsFailed implements Action {
  readonly type = TeamsActionsType.LOAD_TEAMS_FAILED;
  constructor(public payload: any) {}
}

export class SelectTeam implements Action {
  readonly type = TeamsActionsType.SELECT_TEAM;
  constructor(public payload: number) {}
}

export type TeamsActions =
  | LoadTeams
  | LoadTeamsSuccess
  | LoadTeamsFailed
  | SelectTeam;
