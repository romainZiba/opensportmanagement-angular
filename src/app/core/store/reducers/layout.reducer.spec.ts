import { initialState, reducer } from "./layout.reducer";
import * as layoutActions from "../actions";

describe("LayoutReducer", () => {
  describe("undefined action", () => {
    it("should return the default state", () => {
      const action = {} as any;
      const result = reducer(undefined, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe("open sidenav action", () => {
    it("should return sidenav open", () => {
      const action = new layoutActions.OpenSidenav();
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe("close sidenav", () => {
    it("should return closed sidenav", () => {
      const action = new layoutActions.CloseSidenav();
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();
    });
  });
});
