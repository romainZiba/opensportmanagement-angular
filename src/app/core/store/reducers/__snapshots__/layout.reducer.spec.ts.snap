// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutReducer close sidenav should return closed sidenav 1`] = `
Object {
  "showSidenav": false,
}
`;

exports[`LayoutReducer open sidenav action should return sidenav open 1`] = `
Object {
  "showSidenav": true,
}
`;

exports[`LayoutReducer undefined action should return the default state 1`] = `
Object {
  "showSidenav": false,
}
`;
